crates/settlement-clients/starknet/src/tests/test.rs

```diff
@@ -0,0 +1,229 @@
+    ));
+    let address = Felt::from_hex(&env_settings.get_settings_or_panic("STARKNET_PUBLIC_KEY")).unwrap();
+
+    let chain_id = provider.chain_id().await.unwrap();
+    let mut account = SingleOwnerAccount::new(provider, signer, address, chain_id, ExecutionEncoding::New);
+
+    // `SingleOwnerAccount` defaults to checking nonce and estimating fees against the latest
+    // block. Optionally change the target block to pending with the following line:
+    account.set_block_id(BlockId::Tag(BlockTag::Pending));
+    (account, madara_process)
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_deployment(#[future] setup: (SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>, MadaraCmd)) {
+    let (account, _madara_process) = setup.await;
+    let account = Arc::new(account);
+
+    let project_root = Path::new(env!("CARGO_MANIFEST_DIR")).ancestors().nth(3).unwrap();
+    let contract_path = project_root.join("crates/settlement-clients/starknet/src/tests/mock_contracts/target/dev");
+    let sierra_class: SierraClass = serde_json::from_reader(
+        std::fs::File::open(contract_path.join("mock_contracts_Piltover.contract_class.json"))
+            .expect("Could not open sierra class file"),
+    )
+    .expect("Failed to parse SierraClass");
+
+    let compiled_class: CompiledClass = serde_json::from_reader(
+        std::fs::File::open(contract_path.join("mock_contracts_Piltover.compiled_contract_class.json"))
+            .expect("Could not open compiled class file"),
+    )
+    .expect("Failed to parse CompiledClass");
+
+    let flattened_class = sierra_class.clone().flatten().unwrap();
+    let compiled_class_hash = compiled_class.class_hash().unwrap();
+
+    let DeclareTransactionResult { transaction_hash: declare_tx_hash, class_hash: _ } =
+        account.declare_v2(Arc::new(flattened_class.clone()), compiled_class_hash).send().await.unwrap();
+    println!("declare tx hash {:?}", declare_tx_hash);
+
+    let is_success = wait_for_tx_success(&account, declare_tx_hash, Duration::from_secs(2)).await;
+    assert!(is_success, "Declare trasactiion failed");
+
+    let contract_factory = ContractFactory::new(flattened_class.class_hash(), account.clone());
+    let deploy_v1 = contract_factory.deploy_v1(vec![], felt!("1122"), false);
+    let deployed_address = deploy_v1.deployed_address();
+
+    env::set_var("STARKNET_CAIRO_CORE_CONTRACT_ADDRESS", deployed_address.to_hex_string());
+    let InvokeTransactionResult { transaction_hash: deploy_tx_hash } =
+        deploy_v1.send().await.expect("Unable to deploy contract");
+
+    let is_success = wait_for_tx_success(&account, deploy_tx_hash, Duration::from_secs(2)).await;
+    assert!(is_success, "Deploy trasaction failed");
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_settle(#[future] setup: (SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>, MadaraCmd)) {
```

Comment:
we can just keep this test case and remove `test_deployment` right?

Reply:
Removed